core/dive.c

```diff
@@ -3020,30 +3020,38 @@ void taglist_cleanup(struct tag_entry **tag_list)
 	}
 }
 
-int taglist_get_tagstring(struct tag_entry *tag_list, char *buffer, int len)
+char* taglist_get_tagstring(struct tag_entry *tag_list)
 {
-	int i = 0;
-	struct tag_entry *tmp;
-	tmp = tag_list;
-	memset(buffer, 0, len);
+	char *buffer = NULL;
+	char *last_pos = NULL;
+	size_t buffer_size = 0;
+	struct tag_entry *tmp = tag_list;
+	/* Compute buffer size counting extra ", " separator(s) */
 	while (tmp != NULL) {
-		int newlength = strlen(tmp->tag->name);
-		if (i > 0)
-			newlength += 2;
-		if ((i + newlength) < len) {
-			if (i > 0) {
-				strcpy(buffer + i, ", ");
-				strcpy(buffer + i + 2, tmp->tag->name);
-			} else {
-				strcpy(buffer, tmp->tag->name);
-			}
-		} else {
-			return i;
+		if (buffer_size > 0) {
+			buffer_size += 2;
 		}
-		i += newlength;
+		buffer_size += strlen(tmp->tag->name);
 		tmp = tmp->next;
 	}
-	return i;
+	/* Allocate buffer and generate string with ", " separator(s) */
+	if (buffer_size > 0) {
+		buffer = malloc(buffer_size + 1);
+	}
+	if (buffer) {
```

Comment:
```
if (!buffer) {
	fprintf(stderr, "taglist_get_tagstring(): cannot allocate buffer of size: u%\n);
	return NULL;
}
```
this would extract the contents of the branch bellow.